crypto/merklesignature/kats_test.go

```diff
@@ -0,0 +1,124 @@
+// Copyright (C) 2019-2022 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package merklesignature
+
+import (
+	"flag"
+	"fmt"
+	"github.com/algorand/go-algorand/config"
+	"github.com/algorand/go-algorand/protocol"
+	"github.com/algorand/go-algorand/test/partitiontest"
+	"testing"
+)
+
+type mssKat struct {
+	publicKey          []byte
+	ctSignature        []byte
+	ephemeralKey       []byte
+	vcIndex            uint64
+	correspondingRound uint64
+	proofDepth         uint8
+	proofBytes         []byte
+	message            []byte
+}
+
+func sliceToHex(b []byte) string {
```

Comment:
this wouldn't work that great with an empty slice.. wouldn't it ?

Reply:
+1

Reply:
going to use JSON format instead.